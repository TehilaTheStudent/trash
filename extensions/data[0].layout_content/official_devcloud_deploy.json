[
  {
    "componentName": "d-form",
    "id": 1669114667282,
    "category": "form",
    "props": {
      "layout": "vertical",
      "ref": "formRef"
    },
    "children": [
      {
        "componentName": "d-form-item",
        "id": 1669114673142,
        "category": "form",
        "props": {
          "label": "Select Task",
          "field": "SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID",
          "help-tips": "",
          "rules": [
            {
              "required": true,
              "message": "This field cannot be left blank.",
              "trigger": [
                "blur",
                "change"
              ]
            },
            {
              "regex": "",
              "message": "请选择需要调用的任务内容不符合规范",
              "trigger": [
                "blur",
                "change"
              ]
            }
          ],
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "link-select",
            "id": 1669114673141,
            "props": {
              "selectType": "normal",
              "singleOptions": {
                "type": "api",
                "fixedOptions": [
                  {
                    "label": "选项",
                    "value": "1",
                    "defaultChecked": false,
                    "visible": false
                  }
                ],
                "apiOptions": {
                  "url": "/v2/pipeline/getPipelineSubTaskResult",
                  "method": "GET",
                  "params": {
                    "projectId": "${project_id}",
                    "taskType": "deployMan",
                    "jobName": "",
                    "recordNum": "20"
                  },
                  "label": "jobName",
                  "value": "id",
                  "body": {},
                  "responseUrl": "result.jobResultList",
                  "linkedFields": [],
                  "remote": true,
                  "remoteName": "jobName",
                  "remoteQueryField": "params"
                }
              },
              "isUpdateChildAll": true,
              "remote": "",
              "allow-clear": false,
              "filter": "",
              "linkText": "Create One",
              "linkHref": "/deployman/project/${project_id}/deployman/newcreatetemplate",
              "herfTips": "",
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      },
      {
        "componentName": "d-form-item",
        "id": 1669114993360,
        "category": "form",
        "props": {
          "label": "Task parameters",
          "field": "key_1669114993359",
          "help-tips": "",
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "task-parameter",
            "id": 1669114993359,
            "props": {
              "params": {
                "type": "api",
                "apiOptions": {
                  "url": "/v2/pipeline/getPipelineSubTaskParamWithContext",
                  "method": "POST",
                  "params": {
                    "taskType": "deployMan",
                    "subTaskID": "${SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID}",
                    "taskName": "Task_3",
                    "projectId": "${project_id}"
                  },
                  "label": "",
                  "value": "",
                  "body": "",
                  "responseUrl": "result.parameters",
                  "linkedFields": [
                    "SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID"
                  ],
                  "remote": false,
                  "remoteName": "",
                  "remoteQueryField": "params"
                }
              },
              "isCustomVar": true,
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      }
    ]
  }
]