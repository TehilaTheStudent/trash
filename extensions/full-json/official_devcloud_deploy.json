{
  "plugin_name": "official_devcloud_deploy",
  "display_name": "",
  "data": [
    {
      "unique_id": "89f15354f7b34dfc8692e746abf08953",
      "name": "SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID",
      "default_value": "",
      "plugin_name": "official_devcloud_deploy",
      "version": "0.0.1",
      "type": "link-select",
      "workspace_id": "*",
      "layout_content": "[{\"componentName\":\"d-form\",\"id\":1669114667282,\"category\":\"form\",\"props\":{\"layout\":\"vertical\",\"ref\":\"formRef\"},\"children\":[{\"componentName\":\"d-form-item\",\"id\":1669114673142,\"category\":\"form\",\"props\":{\"label\":\"Select Task\",\"field\":\"SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID\",\"help-tips\":\"\",\"rules\":[{\"required\":true,\"message\":\"This field cannot be left blank.\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"\",\"message\":\"请选择需要调用的任务内容不符合规范\",\"trigger\":[\"blur\",\"change\"]}],\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"link-select\",\"id\":1669114673141,\"props\":{\"selectType\":\"normal\",\"singleOptions\":{\"type\":\"api\",\"fixedOptions\":[{\"label\":\"选项\",\"value\":\"1\",\"defaultChecked\":false,\"visible\":false}],\"apiOptions\":{\"url\":\"/v2/pipeline/getPipelineSubTaskResult\",\"method\":\"GET\",\"params\":{\"projectId\":\"${project_id}\",\"taskType\":\"deployMan\",\"jobName\":\"\",\"recordNum\":\"20\"},\"label\":\"jobName\",\"value\":\"id\",\"body\":{},\"responseUrl\":\"result.jobResultList\",\"linkedFields\":[],\"remote\":true,\"remoteName\":\"jobName\",\"remoteQueryField\":\"params\"}},\"isUpdateChildAll\":true,\"remote\":\"\",\"allow-clear\":false,\"filter\":\"\",\"linkText\":\"Create One\",\"linkHref\":\"/deployman/project/${project_id}/deployman/newcreatetemplate\",\"herfTips\":\"\",\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]},{\"componentName\":\"d-form-item\",\"id\":1669114993360,\"category\":\"form\",\"props\":{\"label\":\"Task parameters\",\"field\":\"key_1669114993359\",\"help-tips\":\"\",\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"task-parameter\",\"id\":1669114993359,\"props\":{\"params\":{\"type\":\"api\",\"apiOptions\":{\"url\":\"/v2/pipeline/getPipelineSubTaskParamWithContext\",\"method\":\"POST\",\"params\":{\"taskType\":\"deployMan\",\"subTaskID\":\"${SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID}\",\"taskName\":\"Task_3\",\"projectId\":\"${project_id}\"},\"label\":\"\",\"value\":\"\",\"body\":\"\",\"responseUrl\":\"result.parameters\",\"linkedFields\":[\"SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID\"],\"remote\":false,\"remoteName\":\"\",\"remoteQueryField\":\"params\"}},\"isCustomVar\":true,\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]}]}]",
      "front_data_prop": null,
      "label": null,
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "f41f1d7a85b34e2d89a876abd9c78435",
      "name": "key_1669114993359",
      "default_value": "",
      "plugin_name": "official_devcloud_deploy",
      "version": "0.0.1",
      "type": "task-parameter",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": null,
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    }
  ]
}