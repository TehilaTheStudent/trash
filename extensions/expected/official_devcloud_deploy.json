[
  {
    "key": "relatedPipelineTask",
    "value": "17499339075720215e8e6-1ff8-451a-bd6a-7283aa6e9088.JOB_bpCfY.17499339155850a82af3b-717d-4071-8df6-0401846fe19d"
  },
  {
    "key": "SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID",
    "value": "5ff82d65da4044a8a0cdd8bb8f86ebd9"
  },
  {
    "key": "_OFFICIAL_DEVCLOUD_JOB_NAME_",
    "value": "shell"
  },
  {
    "key": "_OFFICIAL_DEVCLOUD_JOB_APPLICATION_ID_",
    "value": "548cca58c85542d4af4de144b0e61bd5"
  }
]
{
  "key": "relatedPipelineTask",
  "value": "" // this is stagedid.jobid.stepid of a official_devcloud_cloudBuild previous task in the pipeline, only if exists, can leave ""
},
{
  "key": "SYSTEM_DEVCLOUD_SUBPIPELINE_TRIGGER_ID",
  "value": "afe8cc001efb4068b332caf97768bf7f" // this is the arrange_infos.id of the deploy application
},
{
  "key": "_OFFICIAL_DEVCLOUD_JOB_NAME_",
  "value": "production-singapur-manifest-deploment" // this is the name of the deploy application
},
{
  "key": "_OFFICIAL_DEVCLOUD_JOB_APPLICATION_ID_",
  "value": "8186582c52ce48ca8db6130d27d352c2" // this is the id of the deploy application
}
],
